corehq/apps/domain/decorators.py

```diff
@@ -178,6 +179,16 @@ def login_or_basic_ex(allow_cc_users=False):
 login_or_basic = login_or_basic_ex()
 
 
+def login_or_digest_or_basic(fn):
+    @wraps(fn)
+    def _inner(request, *args, **kwargs):
+        return {
+            'basic': login_or_basic_ex(allow_cc_users=True),
+            'digest': login_or_digest_ex(allow_cc_users=True),
+        }[determine_authtype(request)](fn)(request, *args, **kwargs)
```

Comment:
Anywhere that's a more or less "global" file (i.e. a file that's imported indescriminantly to other apps throughout the project). I think even just putting it in the same file (domain/decorators.py) is fine